tools/python/boutiques/tests/test_data_collection.py

```diff
@@ -0,0 +1,155 @@
+from unittest import TestCase
+import datetime as dt
+import os
+import glob
+import mock
+import subprocess
+import pytest
+from boutiques import __file__ as bfile
+from boutiques.localExec import loadJson
+from boutiques_mocks import mock_zenodo_search, MockZenodoRecord
+import boutiques as bosh
+
+def mock_get():
+    mock_record = MockZenodoRecord(1472823, "Example Boutiques Tool", "",
+                                   "https://zenodo.org/api/files/"
+                                   "e5628764-fc57-462e-9982-65f8d6fdb487/"
+                                   "example1_docker.json")
+    return mock_zenodo_search([mock_record])
+
+def retrieve_data_record():
+    data_collect_dict = {}
+    cache_dir = os.path.join(os.path.expanduser('~'), ".cache", "boutiques", "data")
+    if os.path.exists(cache_dir):
+        cache_fls = glob.glob(cache_dir + "/*")
+        if cache_fls:
+            latest_file = max(cache_fls, key=os.path.getctime)
+            path = os.path.join(cache_dir, latest_file)
+            data_collect_dict = loadJson(path)
+    return data_collect_dict
+
+class TestDataCollection(TestCase):
+
+    def get_examples_dir(self):
+        return os.path.join(os.path.dirname(bfile),
+                            "schema", "examples")
+
+    def clean_up(self):
+        # Clean up data collection files
+        cache_dir = os.path.join(os.path.expanduser('~'), ".cache", "boutiques", "data")
+        if os.path.exists(cache_dir):
+            cache_fls = os.listdir(cache_dir)
+            for fl in cache_fls:
+                path = os.path.join(cache_dir, fl)
+                st = os.stat(path)
+                mtime = dt.datetime.fromtimestamp(st.st_mtime)
+                if mtime > dt.datetime.now() - dt.timedelta(minutes=2):
+                    os.remove(path)
+
+    @pytest.mark.skipif(subprocess.Popen("type docker", shell=True).wait(),
+                        reason="Docker not installed")
+    def test_data_collection(self):
+        example1_dir = os.path.join(self.get_examples_dir(), "example1")
+        bosh.execute("launch",
+                        os.path.join(example1_dir,
+                                    "example1_docker.json"),
+                        os.path.join(example1_dir,
+                                    "invocation.json"))
+        data_collect_dict = retrieve_data_record()
+
+        assert data_collect_dict.get("summary") is not None
```

Comment:
use `assertNone`, `assertEqual` and similar functions. See https://github.com/boutiques/boutiques/pull/427